src/components/CampaignInitialisation/CampaignInitialisationScreen.tsx

```diff
@@ -0,0 +1,160 @@
+import React, {
+  FunctionComponent,
+  useEffect,
+  useCallback,
+  useState
+} from "react";
+import { Sentry } from "../../utils/errorTracking";
+import { StyleSheet, View } from "react-native";
+import { size } from "../../common/styles";
+import { NavigationProps } from "../../types";
+import { UpdateByRestartingContent } from "./UpdateByRestartingContent";
+import { UpdateFromAppStoreContent } from "./UpdateFromAppStoreContent";
+import { LoadingView } from "../Loading";
+import { useUpdateCampaignConfig } from "../../hooks/useUpdateCampaignConfig/useUpdateCampaignConfig";
+import { useAuthenticationContext } from "../../context/auth";
+import { useCheckVersion } from "../../hooks/useCheckVersion/useCheckVersion";
+import { useCheckUpdates } from "../../hooks/useCheckUpdates";
+
+const styles = StyleSheet.create({
+  wrapper: {
+    ...StyleSheet.absoluteFillObject,
+    justifyContent: "center",
+    alignItems: "center",
+    paddingHorizontal: size(5)
+  },
+  content: {
+    width: 512,
+    maxWidth: "100%",
+    marginTop: -size(4)
+  }
+});
+
+const RETRY_UPDATE_TIMES = 3;
+
+export const CampaignInitialisationScreen: FunctionComponent<NavigationProps> = ({
```

Comment:
Decided to start a thread so it's easier to see the conversation!!

Questions
1. Are my configurations correct?
2. Are my observations correct for the following steps?

Did this:
* Left the npm start as is
* Changed `/rationally-api/pr71/client/policies` to have no identifiers in SSM (is this supposed to have an impact? Not very sure)
* Changed `/rationally-api/pr71/client/features` to have `"minAppBuildVersion":3`

What I saw:
* On submit of OTP, a loading screen
* The scan ID screen with the warning

```
Warning: Can't perform a React state update on an unmounted component. This is a no-op, but it indicates a memory leak in your application. To fix, cancel all subscriptions and asynchronous tasks in %s.%s, a useEffect cleanup function, 
    in CampaignInitialisationScreen (created by SceneView)
* [native code]:null in __expoConsoleLog
- node_modules/react-native/Libraries/LogBox/LogBox.js:33:4 in console.error
- node_modules/expo/build/environment/muteWarnings.fx.js:27:4 in error
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:608:6 in warningWithoutStack
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:21972:25 in warnAboutUpdateOnUnmountedFiberInDEV
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:19813:40 in scheduleUpdateOnFiber
- node_modules/react-native/Libraries/Renderer/implementations/ReactNativeRenderer-dev.js:11880:16 in dispatchAction
* [native code]:null in dispatchAction
* src/hooks/useUpdateCampaignConfig/useUpdateCampaignConfig.tsx:31:32 in fetchCampaignConfig
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:293:29 in invoke
- node_modules/regenerator-runtime/runtime.js:63:36 in tryCatch
- node_modules/regenerator-runtime/runtime.js:154:27 in invoke
- node_modules/regenerator-runtime/runtime.js:164:18 in PromiseImpl.resolve.then$argument_0
- node_modules/@storybook/addons/node_modules/core-js/modules/es.promise.js:113:24 in microtask$argument_0
- node_modules/@storybook/addons/node_modules/core-js/internals/microtask.js:25:10 in flush
- node_modules/promise/setimmediate/core.js:37:13 in tryCallOne
- node_modules/promise/setimmediate/core.js:123:24 in setImmediate$argument_0
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:135:14 in _callTimer
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:183:16 in _callImmediatesPass
- node_modules/react-native/Libraries/Core/Timers/JSTimers.js:446:30 in callImmediates
* [native code]:null in callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:396:6 in __callImmediates
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:144:6 in __guard$argument_0
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:373:10 in __guard
- node_modules/react-native/Libraries/BatchedBridge/MessageQueue.js:143:4 in flushedQueue
* [native code]:null in flushedQueue
* [native code]:null in invokeCallbackAndReturnFlushedQueue
```

Reply:
Good catch, the issue happens because of this:
1. Enter `CampaignInitialisationScreen`
2. `updateCampaignConfig` is called
3. latest config is fetched, and since it differs, it's saved into async storage + updated context
4. because of this change, `configHashes` is updated
5. `updateCampaignConfig` reference is updated
6. the useEffect in `CampaignInitialisationScreen` triggers `updateCampaignConfig` again
7. at this time, the user has already been navigated to one of the normal screens, and so one of the set states in `CampaignInitialisationScreen` is performed on an already unmounted `CampaignInitialisationScreen`

What I've done is to make an exception to the exhaustive dependencies for this particular `useEffect` as `updateCampaignConfig` should only be called once when this screen is loaded.

Let me know if you know of a better way!

Reply:
Solution sounds good! Tested the same flow and it works now without the warning.